libethashseal/Ethash.cpp

```diff
@@ -152,8 +152,13 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 	{
 		if (_bi.number() >= chainParams().u256Param("EIP158ForkBlock"))
 		{
-			int chainID(chainParams().u256Param("chainID"));
-			_t.checkChainId(chainID);
+			if (_bi.number() >= chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
+				_t.checkChainId(0);
```

Comment:
the issue is that chain params has chainid = 1  
and zero sig transaction has chainid = 0 
meaning that we wont accept zerosig transaction on any chain if it's id != 0